From ee21b6e04b504ea0910b03a12c657dd5174edea1 Mon Sep 17 00:00:00 2001
From: Per Persson <per.persson@intel.com>
Date: Wed, 27 Apr 2016 15:11:23 +0200
Subject: [PATCH] Camera: ov8858: Update BLC

Change MRD Black Level Calibration register values to the values used
in L in order to improve the behavior

Change-Id: I1272be025b49a426861c0d79a96a629d0b166c8b
Orig-Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31380
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-33181
Signed-off-by: Per Persson <per.persson@intel.com>
Reviewed-on: https://android.intel.com:443/501800
Reviewed-on: https://icggerrit.ir.intel.com/121883
Build: Dessert M BTNS (SC)
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Validation-Android: Dessert M BXT RVP (SC)
Approver: Mikael Persson <mikael.persson@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Reviewed-on: https://android.intel.com:443/514613
---
 drivers/external_drivers/camera/drivers/media/i2c/ov8858.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
index 496175d..f56226c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov8858.h
@@ -1293,12 +1293,24 @@ static const struct ov8858_reg ov8858_1640x926[] = {
 };
 
 static const struct ov8858_reg ov8858_BLC_MRD[] = {
+	{OV8858_8BIT, 0x400A, 0x01}, /* BLC CTRL0A */
+	{OV8858_8BIT, 0x400D, 0x10}, /* BLC CTRL0D */
+	{OV8858_8BIT, 0x4011, 0x20}, /* BLC CTRL11 */
 	{OV8858_8BIT, 0x4022, 0x0C}, /* Anchor left start */
 	{OV8858_8BIT, 0x4023, 0x60}, /* Anchor left end */
 	{OV8858_8BIT, 0x4024, 0x0F}, /* Anchor right start */
 	{OV8858_8BIT, 0x4026, 0x0F}, /* Anchor right end */
 	{OV8858_8BIT, 0x402E, 0x04}, /* Bottom black line start = 12 */
 	{OV8858_8BIT, 0x402F, 0x08}, /* Bottom black line number = 2 */
+	{OV8858_8BIT, 0x403E, 0x08}, /* BLC CTRL3E */
+	{OV8858_8BIT, 0x4040, 0x07}, /* BLC CTRL40 */
+	{OV8858_8BIT, 0x4041, 0xc6}, /* BLC CTRL41 */
+	{OV8858_8BIT, 0x4500, 0x58},
+	{OV8858_8BIT, 0x58F8, 0x3D},
+	{OV8858_8BIT, 0x58F4, 0x32},
+	{OV8858_8BIT, 0x470B, 0x28},
+	{OV8858_8BIT, 0x4202, 0x00},
+	{OV8858_8BIT, 0x382D, 0xFF},
 	{OV8858_TOK_TERM, 0, 0}
 };
 
-- 
1.9.1

